libsolutil/LP.h

```diff
@@ -0,0 +1,180 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+#pragma once
+
+#include <libsmtutil/SolverInterface.h>
+
+#include <libsolutil/Common.h>
+
+#include <boost/rational.hpp>
+
+#include <vector>
+#include <variant>
+#include <stack>
+#include <compare>
+
+namespace solidity::util
+{
+
+struct Constraint
```

Comment:
Maybe add a comment explaining what this represents and how it's used? I can't infer that info just by looking at it

Reply:
I will! It's a linear constraint on variables:
- `data[0] = data[1] * x_1 + data[2] * x_2 + ...` (if equality is true)
- `data[0] <= data[1] * x_1 + data[2] * x_2 + ...` (if equality is false)